### ISSUE TITLE
Logical error: 'Trying to get name of not a column: ExpressionList'

### ISSUE THREAD

**Issue by nickitat** [ClickHouse/ClickHouse#43199]
```
[ip-172-31-71-88] 2022.11.12 18:39:25.802487 [ 154 ] {2f7a66f6-e9c8-4410-aa43-cc13b01b2919} <Fatal> : Logical error: 'Trying to get name of not a column: ExpressionList'.
[ip-172-31-71-88] 2022.11.12 18:39:25.803163 [ 416 ] <Fatal> BaseDaemon: ########################################
[ip-172-31-71-88] 2022.11.12 18:39:25.803312 [ 416 ] <Fatal> BaseDaemon: (version 22.11.1.1, build id: 5C63E3023D4F9976505004302B276876F8844A68) (from thread 154) (query_id: 2f7a66f6-e9c8-4410-aa43-cc13b01b2919) (query: SELECT argMax(untuple(untuple((('-0.2147483649', NULL), ('-2' * NULL) * NULL, * EXCEPT b)), x)) FROM (SELECT (number, number + 1) AS x FROM numbers(untuple((('-0.2147483649', NULL), ('0.3' * NULL) * NULL)), 10))) Received signal Aborted (6)
[ip-172-31-71-88] 2022.11.12 18:39:25.803449 [ 416 ] <Fatal> BaseDaemon: 
[ip-172-31-71-88] 2022.11.12 18:39:25.803606 [ 416 ] <Fatal> BaseDaemon: Stack trace: 0x7f3a7d71d00b 0x7f3a7d6fc859 0x1f93a1fb 0x1f93a73b 0x1fdd365d 0x35640dfd 0x31df0e5e 0x2d3ff3d8 0x2d3fe3f3 0x2dd1be6a 0x312358a6 0x312115bd 0x31230de7 0x312115bd 0x3120c0aa 0x31207409 0x314bab52 0x3133f160 0x31e2d6e7 0x31e27460 0x343bc884 0x343ef883 0x3be702ef 0x3be71073 0x3c357f5c 0x3c351268 0x7f3a7d8d4609 0x7f3a7d7f9133
[ip-172-31-71-88] 2022.11.12 18:39:25.803765 [ 416 ] <Fatal> BaseDaemon: 3. raise @ 0x7f3a7d71d00b in ?
[ip-172-31-71-88] 2022.11.12 18:39:25.803897 [ 416 ] <Fatal> BaseDaemon: 4. abort @ 0x7f3a7d6fc859 in ?
[ip-172-31-71-88] 2022.11.12 18:39:25.945021 [ 416 ] <Fatal> BaseDaemon: 5. ./build_docker/../src/Common/Exception.cpp:40: DB::abortOnFailedAssertion(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) @ 0x1f93a1fb in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:26.085042 [ 416 ] <Fatal> BaseDaemon: 6.1. inlined from ./build_docker/../src/Common/Exception.cpp:63: DB::handle_error_code(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, int, bool, std::__1::vector<void*, std::__1::allocator<void*>> const&)
[ip-172-31-71-88] 2022.11.12 18:39:26.085188 [ 416 ] <Fatal> BaseDaemon: 6. ./build_docker/../src/Common/Exception.cpp:70: DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, int, bool) @ 0x1f93a73b in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:26.752442 [ 416 ] <Fatal> BaseDaemon: 7. ./build_docker/../src/Parsers/IAST.h:51: DB::IAST::appendColumnName(DB::WriteBuffer&) const @ 0x1fdd365d in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:26.795559 [ 416 ] <Fatal> BaseDaemon: 8.1. inlined from ./build_docker/../src/IO/WriteBufferFromString.h:37: DB::WriteBufferFromOwnString::str()
[ip-172-31-71-88] 2022.11.12 18:39:26.795683 [ 416 ] <Fatal> BaseDaemon: 8. ./build_docker/../src/Parsers/IAST.cpp:186: DB::IAST::getColumnName() const @ 0x35640dfd in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:26.992450 [ 416 ] <Fatal> BaseDaemon: 9. ./build_docker/../src/Interpreters/evaluateConstantExpression.cpp:74: DB::evaluateConstantExpression(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context const> const&) @ 0x31df0e5e in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:27.083321 [ 416 ] <Fatal> BaseDaemon: 10. ./build_docker/../src/TableFunctions/TableFunctionNumbers.cpp:0: DB::TableFunctionNumbers<false>::evaluateArgument(std::__1::shared_ptr<DB::Context const>, std::__1::shared_ptr<DB::IAST>&) const @ 0x2d3ff3d8 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:27.172559 [ 416 ] <Fatal> BaseDaemon: 11. ./build_docker/../src/TableFunctions/TableFunctionNumbers.cpp:0: DB::TableFunctionNumbers<false>::executeImpl(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context const>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, DB::ColumnsDescription) const @ 0x2d3fe3f3 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:27.331224 [ 416 ] <Fatal> BaseDaemon: 12. ./build_docker/../src/TableFunctions/ITableFunction.cpp:0: DB::ITableFunction::execute(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context const>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, DB::ColumnsDescription, bool) const @ 0x2dd1be6a in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:27.822977 [ 416 ] <Fatal> BaseDaemon: 13. ./build_docker/../src/Analyzer/Passes/QueryAnalysisPass.cpp:5401: DB::(anonymous namespace)::QueryAnalyzer::resolveQueryJoinTreeNode(std::__1::shared_ptr<DB::IQueryTreeNode>&, DB::(anonymous namespace)::IdentifierResolveScope&, DB::(anonymous namespace)::QueryExpressionsAliasVisitor&) @ 0x312358a6 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:28.248138 [ 416 ] <Fatal> BaseDaemon: 14. ./build_docker/../src/Analyzer/Passes/QueryAnalysisPass.cpp:5852: DB::(anonymous namespace)::QueryAnalyzer::resolveQuery(std::__1::shared_ptr<DB::IQueryTreeNode> const&, DB::(anonymous namespace)::IdentifierResolveScope&) @ 0x312115bd in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:28.737713 [ 416 ] <Fatal> BaseDaemon: 15. ./build_docker/../src/Analyzer/Passes/QueryAnalysisPass.cpp:5338: DB::(anonymous namespace)::QueryAnalyzer::resolveQueryJoinTreeNode(std::__1::shared_ptr<DB::IQueryTreeNode>&, DB::(anonymous namespace)::IdentifierResolveScope&, DB::(anonymous namespace)::QueryExpressionsAliasVisitor&) @ 0x31230de7 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:29.171463 [ 416 ] <Fatal> BaseDaemon: 16. ./build_docker/../src/Analyzer/Passes/QueryAnalysisPass.cpp:5852: DB::(anonymous namespace)::QueryAnalyzer::resolveQuery(std::__1::shared_ptr<DB::IQueryTreeNode> const&, DB::(anonymous namespace)::IdentifierResolveScope&) @ 0x312115bd in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:29.571507 [ 416 ] <Fatal> BaseDaemon: 17.1. inlined from ./build_docker/../src/Analyzer/Passes/QueryAnalysisPass.cpp:987: DB::(anonymous namespace)::QueryAnalyzer::resolve(std::__1::shared_ptr<DB::IQueryTreeNode>, std::__1::shared_ptr<DB::IQueryTreeNode> const&, std::__1::shared_ptr<DB::Context const>)
[ip-172-31-71-88] 2022.11.12 18:39:29.571605 [ 416 ] <Fatal> BaseDaemon: 17. ./build_docker/../src/Analyzer/Passes/QueryAnalysisPass.cpp:6176: DB::QueryAnalysisPass::run(std::__1::shared_ptr<DB::IQueryTreeNode>, std::__1::shared_ptr<DB::Context const>) @ 0x3120c0aa in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:29.601944 [ 416 ] <Fatal> BaseDaemon: 18. ./build_docker/../src/Analyzer/QueryTreePassManager.cpp:97: DB::QueryTreePassManager::run(std::__1::shared_ptr<DB::IQueryTreeNode>) @ 0x31207409 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:29.660394 [ 416 ] <Fatal> BaseDaemon: 19.1. inlined from ./build_docker/../src/Analyzer/QueryTreePassManager.h:13: ~QueryTreePassManager
[ip-172-31-71-88] 2022.11.12 18:39:29.660506 [ 416 ] <Fatal> BaseDaemon: 19.2. inlined from ./build_docker/../src/Interpreters/InterpreterSelectQueryAnalyzer.cpp:57: DB::(anonymous namespace)::buildQueryTreeAndRunPasses(std::__1::shared_ptr<DB::IAST> const&, std::__1::shared_ptr<DB::Context const> const&)
[ip-172-31-71-88] 2022.11.12 18:39:29.660558 [ 416 ] <Fatal> BaseDaemon: 19. ./build_docker/../src/Interpreters/InterpreterSelectQueryAnalyzer.cpp:67: DB::InterpreterSelectQueryAnalyzer::InterpreterSelectQueryAnalyzer(std::__1::shared_ptr<DB::IAST> const&, DB::SelectQueryOptions const&, std::__1::shared_ptr<DB::Context const>) @ 0x314bab52 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:29.723375 [ 416 ] <Fatal> BaseDaemon: 20.1. inlined from ./build_docker/../contrib/libcxx/include/__memory/unique_ptr.h:0: std::__1::__unique_if<DB::InterpreterSelectQueryAnalyzer>::__unique_single std::__1::make_unique[abi:v15003]<DB::InterpreterSelectQueryAnalyzer, std::__1::shared_ptr<DB::IAST>&, DB::SelectQueryOptions const&, std::__1::shared_ptr<DB::Context>&>(std::__1::shared_ptr<DB::IAST>&, DB::SelectQueryOptions const&, std::__1::shared_ptr<DB::Context>&)
[ip-172-31-71-88] 2022.11.12 18:39:29.723463 [ 416 ] <Fatal> BaseDaemon: 20. ./build_docker/../src/Interpreters/InterpreterFactory.cpp:134: DB::InterpreterFactory::get(std::__1::shared_ptr<DB::IAST>&, std::__1::shared_ptr<DB::Context>, DB::SelectQueryOptions const&) @ 0x3133f160 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:30.064396 [ 416 ] <Fatal> BaseDaemon: 21. ./build_docker/../src/Interpreters/executeQuery.cpp:0: DB::executeQueryImpl(char const*, char const*, std::__1::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum, DB::ReadBuffer*) @ 0x31e2d6e7 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:30.438168 [ 416 ] <Fatal> BaseDaemon: 22. ./build_docker/../src/Interpreters/executeQuery.cpp:1073: DB::executeQuery(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::shared_ptr<DB::Context>, bool, DB::QueryProcessingStage::Enum) @ 0x31e27460 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:30.693505 [ 416 ] <Fatal> BaseDaemon: 23. ./build_docker/../src/Server/TCPHandler.cpp:0: DB::TCPHandler::runImpl() @ 0x343bc884 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:31.073850 [ 416 ] <Fatal> BaseDaemon: 24. ./build_docker/../src/Server/TCPHandler.cpp:1904: DB::TCPHandler::run() @ 0x343ef883 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:31.083972 [ 416 ] <Fatal> BaseDaemon: 25. ./build_docker/../contrib/poco/Net/src/TCPServerConnection.cpp:57: Poco::Net::TCPServerConnection::start() @ 0x3be702ef in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:31.102479 [ 416 ] <Fatal> BaseDaemon: 26.1. inlined from ./build_docker/../contrib/libcxx/include/__memory/unique_ptr.h:48: std::__1::default_delete<Poco::Net::TCPServerConnection>::operator()[abi:v15003](Poco::Net::TCPServerConnection*) const
[ip-172-31-71-88] 2022.11.12 18:39:31.102584 [ 416 ] <Fatal> BaseDaemon: 26.2. inlined from ./build_docker/../contrib/libcxx/include/__memory/unique_ptr.h:305: std::__1::unique_ptr<Poco::Net::TCPServerConnection, std::__1::default_delete<Poco::Net::TCPServerConnection>>::reset[abi:v15003](Poco::Net::TCPServerConnection*)
[ip-172-31-71-88] 2022.11.12 18:39:31.102642 [ 416 ] <Fatal> BaseDaemon: 26.3. inlined from ./build_docker/../contrib/libcxx/include/__memory/unique_ptr.h:259: ~unique_ptr
[ip-172-31-71-88] 2022.11.12 18:39:31.102688 [ 416 ] <Fatal> BaseDaemon: 26. ./build_docker/../contrib/poco/Net/src/TCPServerDispatcher.cpp:116: Poco::Net::TCPServerDispatcher::run() @ 0x3be71073 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:31.125935 [ 416 ] <Fatal> BaseDaemon: 27. ./build_docker/../contrib/poco/Foundation/src/ThreadPool.cpp:0: Poco::PooledThread::run() @ 0x3c357f5c in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:31.145019 [ 416 ] <Fatal> BaseDaemon: 28. ./build_docker/../contrib/poco/Foundation/src/Thread_POSIX.cpp:0: Poco::ThreadImpl::runnableEntry(void*) @ 0x3c351268 in /workspace/clickhouse
[ip-172-31-71-88] 2022.11.12 18:39:31.145095 [ 416 ] <Fatal> BaseDaemon: 29. ? @ 0x7f3a7d8d4609 in ?
[ip-172-31-71-88] 2022.11.12 18:39:31.145147 [ 416 ] <Fatal> BaseDaemon: 30. clone @ 0x7f3a7d7f9133 in ?
[ip-172-31-71-88] 2022.11.12 18:39:32.025753 [ 416 ] <Fatal> BaseDaemon: Integrity check of the executable skipped because the reference checksum could not be read. (calculated checksum: EFDE47B7CACBC1925FEE1EA12A743BCD)
```

**Comment by alexey-milovidov** [comment]
Fixed.

**Comment by robot-ch-test-poll2** [comment]
### Version info
- Resolved by: #106647
- Merged into: `26.6.1.1016`
